org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/presentation/DSPBreakpointDetailPane.java

```diff
@@ -0,0 +1,330 @@
+			if (adapted instanceof IBreakpoint b) {
+				first = b;
+			}
+		}
+		return first instanceof DSPLineBreakpoint d ? d : null;
+	}
+
+	@Override
+	public Control createControl(final Composite parent) {
+		control = new Composite(parent, SWT.NONE);
+		control.setBackground(UI.getDisplay().getSystemColor(SWT.COLOR_LIST_BACKGROUND));
+		GridLayoutFactory.swtDefaults().numColumns(1).equalWidth(false).applyTo(control);
+		GridDataFactory.fillDefaults().grab(true, true).applyTo(control);
+
+		/*
+		 * column row
+		 */
+		final var colRow = new Composite(control, SWT.NONE);
+		GridLayoutFactory.swtDefaults().numColumns(2).margins(0, 0).equalWidth(false).applyTo(colRow);
+		GridDataFactory.fillDefaults().grab(true, false).applyTo(colRow);
+
+		final var colHint = "Inline breakpoint column (1-based).\nSet > 0 to break at a specific character position on this line; set 0 to use the adapter's default stoppable location.";
+		final var colLabel = new Label(colRow, SWT.NONE);
+		colLabel.setText("Column");
+		colLabel.setToolTipText(colHint);
+		GridDataFactory.swtDefaults().align(SWT.BEGINNING, SWT.CENTER).applyTo(colLabel);
+
+		columnSpinner = new Spinner(colRow, SWT.BORDER);
+		columnSpinner.setMinimum(0);
+		columnSpinner.setMaximum(Integer.MAX_VALUE);
+		columnSpinner.setToolTipText(colHint);
+		GridDataFactory.swtDefaults().hint(100, SWT.DEFAULT).applyTo(columnSpinner);
+
+		/*
+		 * hit condition
+		 */
+		final var hitHint = "Trigger this breakpoint only on specific hit counts.\nExamples: 5 (5th hit), >= 10 (10th and later), % 3 == 0 (every 3rd hit).";
+		final var hitRow = new Composite(control, SWT.NONE);
+		GridLayoutFactory.swtDefaults().numColumns(2).margins(0, 0).equalWidth(false).applyTo(hitRow);
+		GridDataFactory.fillDefaults().grab(true, false).applyTo(hitRow);
+
+		enableHitConditionButton = new Button(hitRow, SWT.CHECK);
+		enableHitConditionButton.setText("Hit condition");
+		enableHitConditionButton.setToolTipText(hitHint);
+		GridDataFactory.swtDefaults().indent(0, 0).align(SWT.BEGINNING, SWT.CENTER).applyTo(enableHitConditionButton);
+
+		hitConditionText = new Text(hitRow, SWT.BORDER);
+		hitConditionText.setToolTipText(hitHint);
+		GridDataFactory.fillDefaults().grab(true, false).applyTo(hitConditionText);
+
+		/*
+		 * condition
+		 */
+		final var condHint = "The breakpoint stops only if the expression evaluates to true in the program context.";
+		enableConditionButton = new Button(control, SWT.CHECK);
+		enableConditionButton.setText("Condition");
+		enableConditionButton.setToolTipText(condHint);
+
+		conditionEditor = SourceCodeEditor.create(control, SWT.NONE);
+		conditionEditor.setToolTipText(condHint);
+		conditionEditor.setEditorLayoutData(GridDataFactory.fillDefaults().grab(true, true).hint(SWT.DEFAULT, 120));
+
+		hookListeners();
+		return control;
+	}
+
+	@Override
+	public void display(final IStructuredSelection selection) {
+		final var selectedBP_ = selectedBP = getSelectedBreakPoint(selection);
```

Comment:
what about prepending `the` for this kind of variables instead of appending `_`? It looks more readable to me.
```suggestion
		final var theSelectedBP = selectedBP = getSelectedBreakPoint(selection);
```